iceoryx_binding_c/include/iceoryx_binding_c/wait_set.h

```diff
@@ -35,56 +35,38 @@ iox_ws_t iox_ws_init(iox_ws_storage_t* self);
 /// @param[in] self the handle which should be deinitialized
 void iox_ws_deinit(iox_ws_t const self);
 
-/// @brief Verifies if a given condition is attached to the WaitSet self
-/// @param[in] self the handle to the WaitSet
-/// @param[in] condition condition where you would like to know if its contained in self
-/// @return true if condition is contained in self, otherwise false
-bool iox_ws_is_condition_attached(iox_ws_t const self, iox_cond_t const condition);
-
-/// @brief Attach a condition to your wait set. If you would like to destroy the condition
-///         you have to detach it first from the wait set.
-/// @param[in] self handle to the wait set
-/// @param[in] condition the condition you would like to attach
-/// @return if the attachment was successful it returns WaitSetResult_SUCCESS, otherwise
-///          an enum which describes the error
-ENUM iox_WaitSetResult iox_ws_attach_condition(iox_ws_t const self, iox_cond_t const condition);
-
-/// @brief detaches a condition.
-/// @param[in] self handle to the wait set
-/// @param[in] condition the condition you would like to detach
-void iox_ws_detach_condition(iox_ws_t const self, iox_cond_t const condition);
-
-/// @brief detaches all conditions
-/// @param[in] self handle to the wait set
-void iox_ws_detach_all_conditions(iox_ws_t const self);
-
-/// @brief waits until a condition was triggered or the timeout was reached
+/// @brief waits until a trigger was triggered or the timeout was reached
 /// @param[in] self handle to the wait set
 /// @param[in] timeout duration how long this method should wait
-/// @param[in] conditionArray preallocated memory to an array of iox_cond_t in which
-///             the conditions which were triggered can be written to
-/// @param[in] conditionArrayCapacity the capacity of the preallocated conditionArray
-/// @param[in] missedElements if the conditionArray has insufficient size the number of missed elements
+/// @param[in] triggerArray preallocated memory to an array of iox_cond_t in which
+///             the triggers which were triggered can be written to
+/// @param[in] triggerArrayCapacity the capacity of the preallocated triggerArray
+/// @param[in] missedElements if the triggerArray has insufficient size the number of missed elements
 ///             which could not be written into the array are stored here
-/// @return number of elements which were written into the conditionArray
+/// @return number of elements which were written into the triggerArray
 uint64_t iox_ws_timed_wait(iox_ws_t const self,
                            struct timespec timeout,
-                           iox_cond_t* const conditionArray,
-                           const uint64_t conditionArrayCapacity,
+                           iox_trigger_state_t const triggerArray,
+                           const uint64_t triggerArrayCapacity,
                            uint64_t* missedElements);
```

Comment:
is this missedElements not a dirty hack? Would it be better to call the error handler if there is not enough space in the triggerArray?

Reply:
It looks for me like a valid C approach. It is like the `errno` approach somehow.

And if there is not enough space we never reach an invalid state which just inform the user that they will not receive the full trigger list and some are lost. But this does not lead to an error since if you miss some triggers you will receive them right after the next call to wait since you did not handle them right away.
If the user does not want this behavior they always can handle this as an error themselves when `missedElements != 0`.

If we would call the error handler here we would also need some error handler c binding. Otherwise the error would be handled in a somehow different language domain where the c developer never has access to.